pkg/token/testdata/ServicePrincipalPoPTokenFromBadPoPClaimsVCR.yaml

```diff
@@ -0,0 +1,174 @@
+---
+version: 2
+interactions:
+    - id: 0
+      request:
+        proto: HTTP/1.1
+        proto_major: 1
+        proto_minor: 1
+        content_length: 0
+        transfer_encoding: []
+        trailer: {}
+        host: login.microsoftonline.com
+        remote_addr: ""
+        request_uri: ""
+        body: ""
+        form: {}
+        headers:
+            User-Agent:
+                - microsoft-authentication-library-for-go/v1.1.1-0.20230811173331-99e839c4c840 (go1.19.6; linux)
```

Comment:
can you add kubelogin identifier to the user agent?

Reply:
@weinong do you mean to makes sure "live" call to msal is setting a kubelogin user-agent or just overwriting the vcr cassette ?

Is this is about real scenario; msal does not expose a helper function to do that. Only way I can think about is to pass a custom httclient with a round tripper. This will requires more code.

Reply:
yes, live calls. huh... azidentity can do that. so i suppose msal can as well? if not possible, then forget about it

Reply:
It's is doable but not as easy as azidentity. IMO this can be a separated PR if really needed.

Reply:
Will resolve this comment as there are no action items